src/Orleans/Configuration/ProviderConfiguration.cs

```diff
@@ -193,23 +193,39 @@ public IList<IProvider> Children
     [Serializable]
     public class ProviderCategoryConfiguration
     {
-        public string Name { get; set; }
+        public ProviderCategoryConfiguration()
+        {
+            Providers = new Dictionary<string, IProviderConfiguration>();
+        }
 
-        public IDictionary<string, IProviderConfiguration> Providers { get; set; }
+        internal ProviderCategoryConfiguration(string name) : this()
```

Comment:
@veikkoeeva 
> We just need to discuss this and establish some kind of a guideline and go with that.

Seems like a good plan.

When to people can't agree on something they should find a trade-off. So:
1. I've removed dictionary extensions (Find) class - so there is nothing to comment anymore.
2. I've reverted `ProviderConfiguration` class to what it was before - so there is no offending internal ctor anymore. Making setters private was a mistake, since it is a breaking change for ppl who might be already using it.